sui-execution/latest/sui-adapter/src/temporary_store.rs

```diff
@@ -559,13 +559,13 @@ impl<'backing> TemporaryStore<'backing> {
                             "Input objects must be address owned, shared, consensus, or immutable"
                         )
                     }
-                    // TODO: Implement support for ConsensusV2 objects.
-                    Owner::ConsensusV2 { .. } => todo!(),
                 }
             })
             .filter(|id| {
                 // remove any non-mutable inputs. This will remove deleted or readonly shared
                 // objects
+                // TODO: Do we want to treat ConsensusV2 objects taken immutably as authenticated?
+                // If so this will have to be changed.
```

Comment:
I'm not sure I fully understand this comment here and I'll need some more context.
For clarity: the "mutable" here refers the specification of mutable versus immutable at the transaction level. For this case, it does not matter how the object will eventually be used during execution (i.e. if it was actually mutated or not). 

We will need to deal with deleted ConsensusV2 objects in a similar way. 
Do we plan on having a transaction argument specification for readonly ConsensusV2 objects?
Are just reusing `ObjectArg::SharedObject` for these (which would imply they will be allowed to be readonly)

Reply:
As of now I am just reusing `ObjectArg::SharedObject` for these, hence the question what to do here in the case where it's specifying a `ConsensusV2` object immutably